pkg/k8shandler/cluster.go

```diff
@@ -472,222 +471,33 @@ func (elasticsearchRequest *ElasticsearchRequest) updateNodeStatus(status api.El
 	return nil
 }
 
-// performCertClusterRestart is required when certs need to be refreshed
-// this is a very high priority action since the cluster may be fractured/unusable
-// in the case where certs aren't all rolled out correctly or are expired
-func (elasticsearchRequest *ElasticsearchRequest) performCertClusterRestart() error {
-	esClient := elasticsearchRequest.esClient
-
-	// make sure we have nodes that are scheduled for full cluster restart first
-	certRedeployNodes := getScheduledCertRedeployNodes(elasticsearchRequest.cluster)
-	clusterStatus := &elasticsearchRequest.cluster.Status
-
-	// 1 -- precheck
-	// no restarting conditions set
-	if len(certRedeployNodes) > 0 {
-
-		if containsClusterCondition(api.Restarting, v1.ConditionFalse, clusterStatus) &&
-			containsClusterCondition(api.UpdatingSettings, v1.ConditionFalse, clusterStatus) {
+// this function should be called before we try doing operations to make sure all our nodes are
+// first ready
+func (elasticsearchRequest *ElasticsearchRequest) ClusterReady() bool {
 
-			// We don't want to gate on cluster health -- we may be in a bad state
-			// because pods aren't all available
-			logrus.Infof("Beginning full cluster restart for cert redeploy on %v", elasticsearchRequest.cluster.Name)
+	// bypass using Status -- check all our pods first, make sure they're all 'Ready'
```

Comment:
What does bypass using status mean here?

Reply:
instead of using the current status field